p4-16/psa/PSA.mdk

```diff
@@ -2572,3 +2543,139 @@ Certainly if there are long durations of time when egress decides to
+egress port, and class of service will pass through the system in FIFO
+order if they follow all recommendations above, unicast packets with
+the same ingress and egress port, but different classes of service,
+may be processed by the egress control block in a different order than
+they were processed by the ingress control block.
+
+If an implementation satisfies recommendations 1 through 3, then
+unicast traffic assigned to the same class of service will maintain
+its relative order through the device.
+
+Recommendation 4: PRE multicast packets (i.e. those that follow the
+"Make 0 or more copies" path in the pseudocode of section
+[#sec-after-ingress]) that arrived on the same ingress port, did
+ingress processing once, were sent to the PRE with the same pair of
+values (`class_of_service`, `multicast_group`), for copies destined to
+the same egress port, and with the same pair of values for
+(`egress_port`, `instance`), should begin egress processing in the
+same relative order as they began ingress processing.
+
+There is no such expectation for multicast packets with different
+`class_of_service` values, again because of separate queues in the PRE
+for different `class_of_service` values.
+
+It is also understood that for a short period of time after the
+control plane modifies the set of copies to be made for a particular
+`multicast_group` value, that it may be especially difficult to
+satisfy Recommendation 4.  That recommendation is intended to apply
+only when the set of copies to be made for the multicast group has
+remained unchanged for a period of time.
+
+If an implementation satisfies recommendations 1, 2, and 4, then
+multicast traffic assigned to the same class of service will maintain
+its relative order through the device, when multicast group membership
+has been stable for long enough.
+
+Note that there is no recommendation to enforce any relative ingress
+to egress processing order of unicast packets vs. multicast packets.
+Commonly used mechanisms for creating multicast copies in the PRE
+allow unicast packets to "go around" the packet replication logic,
+which is unnecessary for unicast packets, and thus change the relative
+order of such packets there.  Also, it is common in packet buffers to
+use separate queues for multicast traffic versus unicast.
+
+
+We give some motivations for these recommendations below.
+
+1. Expectations of hosts
+
+While the Internet Protocol does not have strong ordering requirements
+for sequences of packets sent by one host to another, there are still
+widely deployed implementations of TCP that lead to significantly
+degraded throughput when the network frequently delivers packets to
+the receiver in a different order than they were transmitted.  While
+significant research and development has been done to mitigate this
+issue, e.g. later Linux TCP implementations (since 2011 when version
+2.6.35 was released) are much more resistant to this problem than
```

Comment:
Nit: s/resistant/resilient

Reply:
done.  Resilient is definitely a better word here.